hail/src/main/scala/is/hail/io/fs/FS.scala

```diff
@@ -151,6 +151,15 @@ abstract class FSSeekableInputStream extends InputStream with Seekable {
     toTransfer
   }
 
+  def seek(newPos: Long): Unit = {
+    bb.clear()
+    bb.limit(0)
+    pos = newPos
+    adjustSeekInternalState(newPos)
+  }
```

Comment:
So I think I've slightly changed my mind here about this extra method approach. I think we should just keep seek like it is here without the call to `adjustSeekInternalState`, and then the `GoogleStorageFS` should override `seek` with a call to `super.seek` and whatever additional bookkeeping we're currently putting in `adjustSeekInternalState`.